shell/platform/android/test/io/flutter/embedding/engine/renderer/FlutterRendererTest.java

```diff
@@ -0,0 +1,60 @@
+package io.flutter.embedding.engine.renderer;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import io.flutter.embedding.engine.FlutterJNI;
+import io.flutter.embedding.engine.renderer.FlutterRenderer;
+import io.flutter.embedding.engine.renderer.OnFirstFrameRenderedListener;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+@Config(manifest=Config.NONE)
+@RunWith(RobolectricTestRunner.class)
+public class FlutterRendererTest {
+  FlutterJNITest jniUnderTest;
+  FlutterRenderer rendererUnderTest;
+
+  @Before
+  public void setUp() {
+    jniUnderTest = new FlutterJNITest();
+    rendererUnderTest = new FlutterRenderer(jniUnderTest);
+  }
+
+  @Test
+  public void firstFrameListenerCanRemoveThemselves() {
```

Comment:
I'd like to follow a policy whereby any true unit test prefixes the test with "it", e.g., "itAllowsFirstFrameListenersToRemoveThemselves". This pattern will help ensure that unit tests really are unit tests by grammatically framing in terms of the object under test.

Reply:
Also, this isn't really testing the broader capability of removing oneself as a listener. This is specifically testing the capabilityof removing oneself during a first frame callback. You may want to make that explicit in the test name.

Reply:
ok. Renamed. 

I think the only way to not remove themselves during the first frame would be something like itAllowsFirstFrameListenersToRemoveThemselvesWithoutUsingAHandlerToPostBackADeferredRemoval :D

How's itAllowsFirstFrameListenersToRemoveThemselvesInline?